src/store/selectors/activities/index.ts

```diff
@@ -259,15 +279,76 @@ function addGiveOperationToActivities(
     },
     operations: OrderedMap({ [operation.id]: operation })
   };
-  return [...activities, newActivity];
+  return activities.set(newActivity.id, newActivity);
+}
+
+function combineOperationWithMintActivity(
+  basicIncomeCache: Required<CombineActivitiesCache>["aggregateBasicIncome"],
+  existingActivity: Activity,
+  operation: MintOperation,
+  creatorMember: Member
+): {
+  combinedActivity: Activity;
+  newBasicIncomeCache: typeof basicIncomeCache;
+} {
+  if (existingActivity.content.actors === RAHA_BASIC_INCOME_MEMBER) {
+    throw new Error(
+      "Unexpected: RAHA_BASIC_INCOME_MEMBER was minting a basic income?"
+    );
+  }
+  const { runningTotal, operations } = basicIncomeCache;
+  const newTotal = runningTotal.add(operation.data.amount);
+  const newBasicIncomeCache: typeof basicIncomeCache = {
+    ...basicIncomeCache,
+    operations: operations.push(operation),
+    runningTotal: newTotal
+  };
+
+  const totalMinted: CurrencyValue = {
+    value: newTotal,
+    currencyType: CurrencyType.Raha,
+    role: CurrencyRole.Transaction
+  };
+
+  const combinedActivity: Activity = {
+    ...existingActivity,
+    // use the newest operation's timestamp
+    timestamp:
+      existingActivity.timestamp > operation.created_at
+        ? existingActivity.timestamp
+        : operation.created_at,
+    content: {
+      ...existingActivity.content,
+      description: ["minted a total of", totalMinted, "of basic income."],
+      // TODO: show the most relevant members to the logged in member first, not
+      // just in the order they're found
+      // Only add the actor if they aren't already in the list, to prevent
```

Comment:
We're not actually performing this check, right? It happens automatically b/c we're using a map? I find this comment a bit confusing.